frontend/src/bundles/common/components/user-photo-uploader/user-photo-cropper.tsx

```diff
@@ -0,0 +1,122 @@
+import 'cropperjs/dist/cropper.css';
+
+import React, { useCallback,useRef,useState } from 'react';
+import { type ReactCropperElement } from 'react-cropper';
+import { Cropper } from 'react-cropper';
+import { IconContext } from 'react-icons';
+import { BsSave } from 'react-icons/bs';
+import { FaFileUpload } from 'react-icons/fa';
+import { FaArrowRotateRight } from 'react-icons/fa6';
+import { GrZoomIn , GrZoomOut } from 'react-icons/gr';
+
+import styles from './styles.module.scss';
+
+interface UploadCropperProperties {
+    image: string | ArrayBuffer | null;
+    setImage: React.Dispatch<React.SetStateAction<string | ArrayBuffer | null>>
+    setIsModalOpen: React.Dispatch<React.SetStateAction<boolean>>;
+    setCurrentPhoto: React.Dispatch<React.SetStateAction<string>>;
+    handleImageChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
+}
+
+const UserPhotoCropper: React.FC<UploadCropperProperties> = ({ image, setImage, setIsModalOpen, setCurrentPhoto, handleImageChange }) => {
+    const [croppedImage, setCroppedImage] = useState<string | undefined | null>(null);
```

Comment:
I think we don't need allow this setState function to accept undefined values, preferable to use null

Reply:
Done